### PR TITLE
NHSNLINK-426: Review screen's paging

### PR DESCRIPTION
**Pull request by arianamihailescu**

### CODE REVIEW

File: api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java [hunk 1]
```diff
@@ -116,6 +116,7 @@ private Bundle searchListByCodeData(String system, String value, String date, IG
                 .where(ListResource.IDENTIFIER.exactly().systemAndValues(system, value))
                 .and(ListResource.DATE.exactly().day(date))
                 .returnBundle(Bundle.class)
+                .cacheControl(new CacheControlDirective().setNoCache(true))
```
Comment (seanmcilvenna) [inline api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java:119]: Makes sense. Wonder if there is a way to set this globally for all instances of IGenericClient. I.E. In `BaseController.getFhirStoreClient()`.

File: api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java [hunk 2]
```diff
@@ -125,49 +120,36 @@ private void resolveMeasure(ReportCriteria criteria, ReportContext context) thro
...
+    List<IBaseResource> bundles = FhirHelper.getAllPages(bundle, fhirStoreClient, ctx);
 
     bundles.parallelStream().forEach(bundleResource -> {
       ListResource resource = (ListResource) ctx.newJsonParser().parseResource(ctx.newJsonParser().setPrettyPrint(false).encodeResourceToString(bundleResource));
-      resource.getEntry().parallelStream().forEach(entry -> {
-        String patientId = entry.getItem().getIdentifier().getSystem() +
-                "|" + entry.getItem().getIdentifier().getValue();
-        patientIds.add(patientId);
-      });
+      List<String> patientResourceIds = resource.getEntry().stream().map((patient) -> patient.getItem().getIdentifier().getSystem() + "|" + patient.getItem().getIdentifier().getValue()).collect(Collectors.toList());
```
Comment (seanmcilvenna) [inline api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java:143]: Might consider separating this out onto multiple lines for readability.

File: api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java [hunk 3]
```diff
@@ -428,21 +410,21 @@ public ReportBundle searchReports(Authentication authentication, HttpServletRequ
           if (andCond) {
             url += "&";
           }
-          url += "identifier=" + URLEncoder.encode(identifier, StandardCharsets.UTF_8);
+          url += "identifier=" + Helper.URLEncode(identifier);
```
Comment (seanmcilvenna) [inline api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java:413]: Nice

File: web/src/main/angular/app/review/review.component.ts [hunk 4]
```diff
@@ -81,43 +89,49 @@ export class ReviewComponent implements OnInit {
...
-        }
-        if (this.filter.submittedDate !== null) {
-            let submittedDate = formatDate(this.filter.submittedDate);
-            filterCriteria += `submittedDate=${submittedDate}&`
-        }
-        if (this.filter.submitter !== "Select submitter") {
-            // find the submitter
-            const submitter = this.submitters.find(p => p.name === this.filter.submitter);
-            filterCriteria += `author=${submitter.id}`
+        if (this.filter.bundleId == "") {
```
Comment (seanmcilvenna) [inline web/src/main/angular/app/review/review.component.ts:92]: Should rename `bundleId` to `reportTypeId` at some point. Doesn't have to be part of this ticket...

File: web/src/main/angular/app/review/review.component.ts [hunk 5]
```diff
@@ -143,12 +157,21 @@ export class ReviewComponent implements OnInit {
...
+        }
+        this.page = newPage;
+        await this.searchReports();
     }
 
     async ngOnInit() {
         await this.searchReports();
         this.measures = await this.reportDefinitionService.getReportDefinitions();
         this.submitters = await this.reportService.getSubmitters();
+        console.log(this.submitters);
```
Comment (seanmcilvenna) [inline web/src/main/angular/app/review/review.component.ts:175]: Remove

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java (hunk 2 above)
Comment (arianamihailescu) [reply api/src/main/java/com/lantanagroup/link/api/controller/ReportController.java]: Done

File: web/src/main/angular/app/review/review.component.ts (hunk 4 above)
Comment (arianamihailescu) [reply web/src/main/angular/app/review/review.component.ts]: Done

File: web/src/main/angular/app/review/review.component.ts (hunk 5 above)
Comment (arianamihailescu) [reply web/src/main/angular/app/review/review.component.ts]: Done

File: api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java (hunk 1 above)
Comment (arianamihailescu) [reply api/src/main/java/com/lantanagroup/link/api/controller/PatientIdentifierController.java:119]: Will research

**Pull request merged**